components/script/dom/gpudevice.rs

```diff
@@ -680,6 +679,145 @@ impl GPUDeviceMethods for GPUDevice {
             true,
         )
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createrenderpipeline
+    fn CreateRenderPipeline(
+        &self,
+        descriptor: &GPURenderPipelineDescriptor,
+    ) -> DomRoot<GPURenderPipeline> {
+        let mut valid = descriptor.parent.layout.is_valid();
+        valid &= descriptor.colorStates.len() <= 4;
+        if descriptor.alphaToCoverageEnabled {
+            valid &= descriptor.sampleCount > 1;
+        }
+
+        let vertex_module = descriptor.vertexStage.module.id().0;
+        let vertex_entry_point = descriptor.vertexStage.entryPoint.to_string();
+        let (fragment_module, fragment_entry_point) = match descriptor.fragmentStage {
+            Some(ref frag) => (Some(frag.module.id().0), Some(frag.entryPoint.to_string())),
+            None => (None, None),
+        };
+
+        let primitive_topology = match descriptor.primitiveTopology {
+            GPUPrimitiveTopology::Point_list => wgt::PrimitiveTopology::PointList,
+            GPUPrimitiveTopology::Line_list => wgt::PrimitiveTopology::LineList,
+            GPUPrimitiveTopology::Line_strip => wgt::PrimitiveTopology::LineStrip,
+            GPUPrimitiveTopology::Triangle_list => wgt::PrimitiveTopology::TriangleList,
+            GPUPrimitiveTopology::Triangle_strip => wgt::PrimitiveTopology::TriangleStrip,
+        };
+
+        let ref rs_desc = descriptor.rasterizationState;
+        let rasterization_state = wgt::RasterizationStateDescriptor {
+            front_face: match rs_desc.frontFace {
+                GPUFrontFace::Ccw => wgt::FrontFace::Ccw,
+                GPUFrontFace::Cw => wgt::FrontFace::Cw,
+            },
+            cull_mode: match rs_desc.cullMode {
+                GPUCullMode::None => wgt::CullMode::None,
+                GPUCullMode::Front => wgt::CullMode::Front,
+                GPUCullMode::Back => wgt::CullMode::Back,
+            },
+            depth_bias: rs_desc.depthBias,
+            depth_bias_slope_scale: *rs_desc.depthBiasSlopeScale,
+            depth_bias_clamp: *rs_desc.depthBiasClamp,
+        };
+
+        let color_states = descriptor
+            .colorStates
+            .iter()
+            .map(|state| wgt::ColorStateDescriptor {
+                format: wgt::TextureFormat::Rgba8UnormSrgb, //TODO: Update this after implementing Texture
+                alpha_blend: create_blend_descriptor(&state.alphaBlend),
+                color_blend: create_blend_descriptor(&state.colorBlend),
+                write_mask: wgt::ColorWrite::from_bits(state.writeMask).unwrap(),
+            })
+            .collect::<Vec<_>>();
```

Comment:
```
error[E0277]: the trait bound `wgt::ColorStateDescriptor: arrayvec::array::Array` is not satisfied in `arrayvec::ArrayVec<wgt::ColorStateDescriptor>`
   --> components/webgpu/lib.rs:110:9
    |
110 |         color_states: ArrayVec<wgt::ColorStateDescriptor>,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `arrayvec::ArrayVec<wgt::ColorStateDescriptor>`, the trait `arrayvec::array::Array` is not implemented for `wgt::ColorStateDescriptor`
    |
    = note: required because it appears within the type `arrayvec::ArrayVec<wgt::ColorStateDescriptor>`
    = note: no field of an enum variant may have a dynamically sized type
```

Reply:
It should be `ArrayVec<[wgt::ColorStateDescriptor; MAX_COLOR_TARGETS]>`